src/store/storeConfig.ts

```diff
@@ -629,7 +629,7 @@ export const storeConfig = (kumaApi: KumaApi): StoreOptions<State> => {
               title: lastSubscription.version.kumaCp.version,
               data: 1,
               route: {
-                name: 'zones',
+                name: 'zone-list-view',
```

Comment:
No action here really unless you want to, and kinda thinking out loud.  But, I quite like `zones` meaning "the collection of `zones`" due to it being plural. I know its a `-view` because its a view component, it probably can't be anything other than a `view` so "view" is a bit redundant, and I know its a `-list` (or `-table`, or `-grid` a.k.a. `-collection`) because of the pluralization.

---

It would be nice to eventually end up with route names being of the form `mesh.zones.show.config` seeing as the visiting this route is keyed by `mesh`, `zones`, `single-zone-identifier`, `config`, i.e. at the `mesh.zones.show.config` route you would see information specific to the `config` of the `single-zone` within all `zones` in a `mesh`. Chances are this is all nested/child-routed in some way.

Just to be clear, not necessary to amend this at all right now, just sharing thoughts.

Reply:
> It would be nice to eventually end up with route names being of the form mesh.zones.show.config seeing as the visiting this route is keyed by mesh, zones, single-zone-identifier, config, i.e. at the mesh.zones.show.config route you would see information specific to the config of the single-zone within all zones in a mesh. Chances are this is all nested/child-routed in some way.

I think this specifically is to be avoided. It makes changing routing hierarchies unnecessarily hard because if you move a route to a hierarchically different position, you now end up with a misleading route name. And you can rename it, but that defeats the purpose of targeting routes by name and not by path: to allow you to move routes and change paths without breaking existing route targets in the process.

In that sense, these names are deliberately chosen to avoid ambiguity. And while it might seem redundant to call the list view for zones "zone list view" (etc.), it’s much nicer to search the project for references to it (one could improve this further by only using constants instead of string literals): a search for `zones` (for example) is much more likely to yield false positives when searching for route names. That’s nearly impossible when using the naming convention I established for routes.

Reply:
WeIl suppose that leads to another bugbear of mine. Generally I prefer to use the word `items` for variables instead of `zones` (you know an `item` is a `Zone` as its the Zone component, change the name of the component to Area and `item` can still be `item` as it now refers to an `Area`). This way the term `zones` is probably only ever used for string routeNames. But in the absence of that, I think its maybe more likely that the inclusion of a `-` makes the searching whilst excluding javascript varnames easier

> I think this specifically is to be avoided. It makes changing routing hierarchies unnecessarily hard because if you move a route to a hierarchically different position, you now end up with a misleading route name. And you can rename it, but that defeats the purpose of targeting routes by name and not by path: to allow you to move routes and change paths without breaking existing route targets in the process.

What happens when you have two routes using the same view i.e. say a service detail page on its own "service-detail-view" and another service detail page alongside the service listing (say in a drawer as a nested route). They can't both be named "service-detail-view" can they? One of them is inside `services.` and one isn't. I also think that possibly the idea of "relative route names" would help with what you explain there.

Anyway I can see there is a large difference of opinion here so glad I asked so I know but happy to leave this one here for now at least.

(just incase, nothing to be done here, just trading ideas)

Reply:
> What happens when you have two routes using the same view i.e. say a service detail page on its own "service-detail-view" and another service detail page alongside the service listing (say in a drawer as a nested route). They can't both be named "service-detail-view" can they? One of them is inside services. and one isn't. I also think that possibly the idea of "relative route names" would help with what you explain there.

That could be called `service-summary-view`, for example.